lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -1976,10 +1976,20 @@ SwiftExpressionParser::Parse(DiagnosticManager &diagnostic_manager,
 
   // If IRGen failed without errors, the root cause may be a fatal
   // Clang diagnostic.
-  if (expr_diagnostics->HasErrors() ||
-      m_swift_ast_ctx.HasClangImporterErrors()) {
-    diagnostic_manager.PutString(eDiagnosticSeverityRemark,
-                                 "couldn't IRGen expression");
+  using ErrorKind = SwiftASTContext::ScopedDiagnostics::ErrorKind;
+  std::optional<ErrorKind> error_kind = expr_diagnostics->GetErrorKindIfAny();
```

Comment:
Maybe add a comment here that is something like:
// GetErrorKindIfAny() returns all diagnostics that occurred to during the lifetime of expr_diagnostics, but there could be earlier ClangImporter errors that still caused the expression to fail.